frontend/public/components/overview.jsx

```diff
@@ -0,0 +1,353 @@
+  return _.filter(resources, function (resource) {
+    const {ownerReferences} = resource.metadata;
+    return _.some(ownerReferences, {
+      uid,
+      controller: true
+    });
+  });
+};
+
+const OverviewToolbar = ({groupOptions, handleFilterChange, handleGroupChange, selectedGroup}) =>
+  <Toolbar className="overview-toolbar">
+    <Toolbar.RightContent>
+      <div className="form-group overview-toolbar__form-group">
+        <label className="overview-toolbar__label">
+          Group by label
+        </label>
+        <Dropdown
+          className="overview-toolbar__dropdown"
+          disabled={_.isEmpty(groupOptions)}
+          items={groupOptions}
+          onChange={handleGroupChange}
+          style={{display: 'inline-block'}}
+          title={selectedGroup}
+        />
+      </div>
+      <div className="form-group overview-toolbar__form-group">
+        <TextFilter
+          label="Resources by name"
+          onChange={handleFilterChange}
+        />
+      </div>
+    </Toolbar.RightContent>
+  </Toolbar>;
+
+OverviewToolbar.displayName = 'OverviewToolbar';
+
+OverviewToolbar.propTypes = {
+  groupOptions: PropTypes.object.isRequired,
+  handleFilterChange: PropTypes.func.isRequired,
+  handleGroupChange: PropTypes.func.isRequired,
+  selectedGroup: PropTypes.string.isRequired
+};
+
+class OverviewDetails extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleFilterChange = this.handleFilterChange.bind(this);
+    this.handleGroupChange = this.handleGroupChange.bind(this);
+    this.clearFilter = this.clearFilter.bind(this);
+
+    this.state = {
+      filterValue: '',
+      items: [],
+      filteredItems: [],
+      groupedItems: [],
+      groupOptions: {},
+      selectedGroupLabel: 'app'
+    };
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    if (this.props.namespace !== prevProps.namespace
+      || this.props.replicationControllers !== prevProps.replicationControllers
+      || this.props.pods !== prevProps.pods
+      || this.props.deploymentConfigs !== prevProps.deploymentConfigs) {
+      this.createOverviewData();
+    } else if (this.state.filterValue !== prevState.filterValue) {
+      const filteredItems = this.filterItems(this.state.items);
+      this.setState({
+        filteredItems,
+        groupedItems: this.groupByLabel(filteredItems, this.state.selectedGroupLabel)
+      });
+    } else if (this.state.selectedGroupLabel !== prevState.selectedGroupLabel) {
+      this.setState({
+        groupedItems: this.groupByLabel(this.state.filteredItems, this.state.selectedGroupLabel)
+      });
+    }
+  }
+
+  sortByDeploymentVersion(replicationControllers, descending) {
+    const compareDeployments = (left, right) => {
+      const leftVersion = parseInt(_.get(left, 'openshift.io/deployment-config.latest-version'), 10);
+      const rightVersion = parseInt(_.get(right, 'openshift.io/deployment-config.latest-version'), 10);
```

Comment:
Doesn't this need to be checking the annotations? Looks like the `_.get` call is wrong.